### PR TITLE
Block locked users

### PR DESCRIPTION
**Pull request by chicocvenancio**

### CODE REVIEW

File: paws/values.yaml [hunk 1]
```diff
@@ -147,14 +147,32 @@ jupyterhub:
...
+
           class Auth(MWOAuthenticator):
               enable_auth_state = True
               def normalize_username(self, username):
                   return username
           
+              async def refresh_user(self, user, handler=None):
+                  client = AsyncHTTPClient()
+                  try:
+                      response = await client.fetch(f"https://meta.wikimedia.org/w/api.php?action=query&format=json&meta=globaluserinfo&guiuser={user.name}")
```
Comment (supertassu) [inline paws/values.yaml:165]: This should probably set a user-agent per https://meta.wikimedia.org/wiki/User-Agent_policy

nit: I'd recommend setting `formatversion=2` in the request, makes the response sometimes a bit nicer to work with

Comment (chicocvenancio) [reply paws/values.yaml]: Nice one with `formatversion=2`, @supertassu allows for a more readable check on the next line.

**Comment by chicocvenancio** [comment]
Regarding UA, this will now use UA for the lock requests, but oauth is still using the default one for mwoauth, and we should probably send a PR to https://github.com/jupyterhub/oauthenticator to have it be configurable.

**Review by supertassu — APPROVED** [review]
lgtm, however we might want to add logging in case we stop a server for a locked account

**Comment by chicocvenancio** [comment]
> we might want to add logging in case we stop a server for a locked account
The base classes seem to do enough logging:

Mocking my user to be locked, the logs look like 

```
[I 2021-11-28 18:28:20.992 JupyterHub spawner:2620] Deleting pod paws-dev/jupyter--43hicocvenancio
[W 2021-11-28 18:28:21.014 JupyterHub base:332] User Chicocvenancio has stale auth info. Login is required to refresh.
[W 2021-11-28 18:28:21.015 JupyterHub log:189] 403 POST /hub/api/users/Chicocvenancio/activity (@172.17.0.1) 513.52ms
[W 2021-11-28 18:28:47.772 JupyterHub app:2131] User Chicocvenancio server stopped with exit code: 1
[I 2021-11-28 18:28:47.772 JupyterHub proxy:309] Removing user Chicocvenancio from proxy (/user/Chicocvenancio/)
```

**Pull request merged**